hazelcast/test/src/sql_test.cpp

```diff
@@ -505,6 +866,392 @@ TEST_F(SqlTest, statement_with_params)
     EXPECT_EQ("-42.73", rows[0].get_object<std::string>(1).value());
 }
 
+// These tests ported from python-client, sql_test.py
+TEST_F(SqlTest, test_execute)
+{
+    create_mapping();
+    auto expecteds = populate_map(11);
+
+    auto result =
+      client.get_sql()
+        .execute((boost::format("SELECT * FROM %1%") % map_name).str())
+        .get();
+
+    for_each_row(result,
+                 assert_row_count{ expecteds.size() },
+                 assert_key_values<int>{ expecteds });
+}
+
+TEST_F(SqlTest, test_execute_with_params)
+{
+    create_mapping();
+    auto expecteds = populate_map(13);
+
+    auto result =
+      client.get_sql()
+        .execute(
+          (boost::format("SELECT * FROM %1% WHERE this > ?") % map_name).str(),
+          6)
+        .get();
+
+    auto expected_row_count = count_if(
+      begin(expecteds), end(expecteds), [](int value) { return value > 6; });
+
+    for_each_row(result,
+                 assert_row_count(expected_row_count),
+                 assert_key_values<int>{ expecteds });
+}
+
+TEST_F(SqlTest, test_execute_with_mismatched_params_when_sql_has_more)
+{
+    create_mapping();
+    (void)populate_map();
+
+    auto execution = client.get_sql().execute(
+      (boost::format("SELECT * FROM %1% WHERE __key > ? AND this > ?") %
+       map_name)
+        .str(),
+      5);
+
+    EXPECT_THROW(execution.get(),
+                 hazelcast::client::sql::hazelcast_sql_exception);
```

Comment:
Can we also verify the returned code is as expected in addition to exception type?